lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -325,10 +325,14 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
         cmd += " --socket=%s" % shlex_quote(socket)
     else:
         cmd += " --host=%s --port=%i" % (shlex_quote(host), port)
+
     if all_databases:
         cmd += " --all-databases"
-    else:
+    elif len(db_name) > 1:
         cmd += " --databases {0}".format(' '.join(db_name))
+    else:
```

Comment:
I don't get the point of letting `db_name` being empty.  
@Andersson007 would you mind explaining the main idea of these changes ?

Reply:
@bmalynovytch i'm not sure i understand you properly but, i think, regarding the code it can't be empty
because of ```name=dict(type='list', required=True, aliases=['db']),```. So there always will be at least one db name in the list.
The main idea is to partially revert https://github.com/ansible/ansible/commit/44058e942588425197208ce7976559d9a49b86d1#diff-4eeda2d2fc550548c262c40b0fbdb808R172
to be able to pass just a single db name without any flags as it was before the commit.

Folks confirmed https://github.com/ansible/ansible/issues/65351 that it works ok.